src/v/security/jwt.h

```diff
@@ -526,11 +529,18 @@ make_verifiers(jwks const& jwks, CryptoPP::AutoSeededRandomPool& rng) {
     auto keys = jwks.keys();
     verifiers vs;
     for (auto const& key : keys) {
-        // Alg is required
-        auto alg = detail::string_view(key, "alg");
-        if (!alg) {
-            return errc::jwk_invalid;
-        }
+        // NOTE(oren): 'alg' field is optional per RFC 7517
+        // https://datatracker.ietf.org/doc/html/rfc7517#section4.4
+        // In particular, Azure doesn't include it, so in its absence we can
+        // just try to verify as rs256 by default.
```

Comment:
I thought I might have been too strict here. I wonder if it's worth doing anything clever with [id_token_signing_alg_values_supported](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata:~:text=and%20public.-,id_token_signing_alg_values_supported,-REQUIRED.%20JSON%20array)?
1. Hope it is a single value (and the list MUST include `RS256`) 
2. Make generating the key lazy and hope that the JWT header has a `kid` and `alg`, that will help tie  these together?

I think 1. will suffice for now, since this is what you've done below, and seems to be the case for the tested providers so far.

Reply:
I guess the upshot here is that we only support situations where the JWT includes both `alg` and a `kid` that indexes into an already-known JWKS s.t. `jwks[kid]` passes RS256 verification. In that sense, we don't lose much from implementing (1) given that we currently only support RS256.

To support other configurations (inline JWK, link to JWKS in JWT header), we would probably want to do something like (2), but I don't know how critical that is.

Reply:
1. Shall suffice for now.